server/text_generation_server/models/idefics.py

```diff
@@ -90,7 +89,16 @@ def __init__(
             weights_loader=weights_loader,
         )
 
-        model = IdeficsForVisionText2Text(config, weights)
+        if config.model_type == "idefics":
+            model = IdeficsForVisionText2Text(config, weights)
+        elif config.model_type == "mllama":
+            model = MllamaForConditionalGeneration(
+                prefix="", config=config, weights=weights
+            )
+        else:
+            raise RuntimeError(f"Unsupported model type {config.model_type}")
```

Comment:
not sure if we want to update the name of this class from `IDEFICSSharded` to something like `VLMShared` since it seems that mllama will use this path too

Reply:
No it doesn't this is old code that needs to be removed, I just fused `idefics.py` and `idefics_causal_lm.py`.